File: netlogo-gui/src/main/prim/_setbreedvariable.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.api.LogoException;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.RuntimePrimitiveException;

public final strictfp class _setbreedvariable
    extends Command {
  private final String name;

  public _setbreedvariable(_breedvariable original) {
    name = original.name();
    this.switches = true;
  }

  @Override
  public String toString() {
    return super.toString() + ":" + name;
  }

  @Override
  public void perform(final Context context) throws LogoException {
    perform_1(context, args[0].report(context));
  }

  public void perform_1(final org.nlogo.nvm.Context context, Object arg0) {
    try {
      context.agent.setBreedVariable(name, arg0);
    } catch (org.nlogo.api.AgentException ex) {
      throw new RuntimePrimitiveException(context, this, ex.getMessage());
    }
    context.ip = next;
  }
}

File: netlogo-headless/src/main/prim/_setbreedvariable.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.AgentException
import org.nlogo.nvm.{ Command, Context}
import org.nlogo.nvm.RuntimePrimitiveException

class _setbreedvariable(name: String) extends Command {

  def this(original: _breedvariable) = this(original.name)

  switches = true

  override def toString =
    super.toString + ":" + name

  override def perform(context: Context): Unit = {
    val value = args(0).report(context)
    try context.agent.setBreedVariable(name, value)
    catch { case ex: AgentException =>
      throw new RuntimePrimitiveException(context, this, ex.getMessage) }
    context.ip = next
  }

}
